packages/pacman.rb

```diff
@@ -0,0 +1,73 @@
+require 'buildsystems/meson'
+
+class Pacman < Meson
+  description 'A library-based package manager with dependency support.'
+  version '6.0.2-1'
+  homepage 'http://www.archlinux.org/pacman/'
+  license 'GPL 2'
+  compatibility 'all'
+  source_url 'https://sources.archlinux.org/other/pacman/pacman-6.0.2.tar.xz'
+  source_sha256 '7d8e3e8c5121aec0965df71f59bedf46052c6cf14f96365c4411ec3de0a4c1a5'
+
+  binary_url({
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/pacman/6.0.2-1_armv7l/pacman-6.0.2-1-chromeos-armv7l.tar.zst',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/pacman/6.0.2-1_armv7l/pacman-6.0.2-1-chromeos-armv7l.tar.zst',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/pacman/6.0.2-1_i686/pacman-6.0.2-1-chromeos-i686.tar.zst',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/pacman/6.0.2-1_x86_64/pacman-6.0.2-1-chromeos-x86_64.tar.zst'
+  })
+  binary_sha256({
+    aarch64: 'f948bca6d9e75b56faabf2c4e178c62e98e44c953d8b5c6d02a96165107e16f2',
+     armv7l: 'f948bca6d9e75b56faabf2c4e178c62e98e44c953d8b5c6d02a96165107e16f2',
+       i686: '9005be863f9d80afa9af3f5dc3843cb8ba04aefdb7108da2d1a2b011e1923ed8',
+     x86_64: '7368961442770824e8753c591d1a604120b5410e61494afe31e5b10b5e0fd242'
+  })
+
+  depends_on 'asciidoc' => :build
+  depends_on 'bash' # pacman demands a bash newer than 4.0
+  depends_on 'curl'
+  depends_on 'fakeroot'
+  depends_on 'gcc_lib' # R
+  depends_on 'glibc'
+  depends_on 'gpgme'
+  depends_on 'libarchive'
+  depends_on 'openssl' # R
+  depends_on 'py3_setuptools' => :build # needed for Python 3.12 compatibility
+  depends_on 'xzutils'
+
+  meson_options "-Droot-dir=#{CREW_PREFIX} \
+             -Dsysconfdir=#{CREW_PREFIX}/etc \
+             -Dlocalstatedir=#{CREW_PREFIX}/var \
+             -Dsharedstatedir=#{CREW_PREFIX}/var/local/lib \
+             -Dmakepkg-template-dir=#{CREW_PREFIX}/share/makepkg-template \
+             -Dscriptlet-shell=#{CREW_PREFIX}/bin/bash \
+             -Dldconfig=/sbin/ldconfig \
+             -Dpkg-ext=.pkg.tar.xz \
+             -Dsrc-ext=.src.tar.gz"
+
+  def self.postinstall
+    # Enable mirror list.
+    open("#{CREW_PREFIX}/etc/pacman.conf", 'a') do |f|
+      f.puts '[core]'
+      f.puts "Server = https://mirrors.kernel.org/archlinux/\$repo/os/\$arch"
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+      f.puts '[extra]'
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+      f.puts '[community]'
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+    end
+    # Figure out how to enable trusts for the gpg sigs...
+    system "sed -i '/#SigLevel/a SigLevel = TrustAll' #{CREW_PREFIX}/etc/pacman.conf"
+    puts 'Installing pacman mirrorlist.'.lightblue
+    system 'curl -Lf https://archlinux.org/mirrorlist/all/ -o mirrorlist'
```

Comment:
Use downloader with sha256 checksum?

Reply:
The mirrorlist will change, but I have changed this to use downloader.